Source/NSDockTile.m

```diff
@@ -90,13 +108,112 @@ - (NSString *) badgeLabel
 
 - (void) setBadgeLabel: (NSString *)label
 {
+  NSImage *tempImage;
+  NSImageRep *rep;
+
   ASSIGNCOPY(_badgeLabel, label);
+
+  tempImage = [[NSImage alloc] initWithSize: [_appIconImage size]];
+
+  rep = [[NSCustomImageRep alloc] initWithDrawSelector: @selector(draw) delegate: self];
+  [rep setSize: [_appIconImage size]];
+  [tempImage addRepresentation: rep];
+  RELEASE(rep);
+
+  [NSApp setApplicationIconImage: tempImage];
 }
 
 - (void) display
 {
   [_contentView setNeedsDisplay: YES];
+
+  if (_showsApplicationBadge)
+    {
+    }
 }
 
-@end
+- (void)draw
+{
+  [_appIconImage compositeToPoint: NSZeroPoint operation: NSCompositeCopy];
 
+  if (_showsApplicationBadge && _badgeLabel)
+    {
+      NSMutableDictionary *attrs;
+      NSPoint textLocation;
+      NSRect discRect;
+      NSSize discSize;
+      NSSize textSize;
+      int pad;
+      NSBezierPath *p;
+      NSPoint point;
+      CGFloat radius;
+      NSSize imageSize;
+      NSColor *badgeBackColor;
+      NSColor *badgeDecorationColor;
+      NSColor *badgeTextColor;
+      NSString *displayString;
+
+      badgeBackColor = [NSColor redColor];
+      badgeDecorationColor = [NSColor lightGrayColor];
+      badgeTextColor = [NSColor whiteColor];
+
+      displayString = _badgeLabel;
+      if ([_badgeLabel length] > 5)
+	displayString = [NSString stringWithFormat:@"%@\u2026%@",[_badgeLabel substringToIndex:2],  [_badgeLabel substringFromIndex:[_badgeLabel  length]-2]];
```

Comment:
Distributing the spaces more evenly would help :-)

Reply:
sorry, mistyped·... but a hint about unicode ellispis?